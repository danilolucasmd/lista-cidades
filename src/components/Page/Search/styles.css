:local(.component) input {
  margin: 40px 40px; }
