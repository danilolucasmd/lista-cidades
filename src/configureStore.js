import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import app from './reducers';

const configureStore = () =>
  createStore(
    app,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
    applyMiddleware(thunk)
  );

export default configureStore;
