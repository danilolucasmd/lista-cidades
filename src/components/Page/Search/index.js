import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import { changeFilters } from 'actionCreators/filters';

import styles from './styles.css';

class Search extends Component {
  constructor(props) {
    super(props);

    this.state = {
      city: '',
      state: '',
    };
  }

  handleSubmit = e => {
    e.preventDefault();

    this.props.changeFilters(this.state);
  }

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  render() {
    return (
      <form className={styles.component} onSubmit={this.handleSubmit}>
        <label>Cidade:</label>
        <input
          id='city'
          name='city'
          value={this.state.city}
          onChange={this.handleChange}
        />

        <label>Estado:</label>
        <input
          id='state'
          name='state'
          value={this.state.state}
          onChange={this.handleChange}
        />

        <button type='submit'>Buscar</button>
      </form >
    );
  }
}

Search.propTypes = {
  changeFilters: PropTypes.func,
};

export default connect(undefined, { changeFilters })(Search);