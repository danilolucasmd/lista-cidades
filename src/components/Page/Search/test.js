import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import Search from './index';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'configureStore';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const store = configureStore();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Search store={store} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should render without throwing an error', () => {
  expect(shallow(<Search store={store} />).exists(<form></form>)).toBe(true);
});

it('renders correctly', () => {
  const tree = renderer.create(
    <Search store={store} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});