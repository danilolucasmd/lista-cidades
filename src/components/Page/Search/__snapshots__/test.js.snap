// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className={undefined}
  onSubmit={[Function]}
>
  <label>
    Cidade:
  </label>
  <input
    id="city"
    name="city"
    onChange={[Function]}
    value=""
  />
  <label>
    Estado:
  </label>
  <input
    id="state"
    name="state"
    onChange={[Function]}
    value=""
  />
  <button
    type="submit"
  >
    Buscar
  </button>
</form>
`;
